docs/guides/collision-handling.md

```diff
@@ -0,0 +1,122 @@
+## Collision handling
+
+Implementing the slices pattern can introduce the risk of name or action collisions, which can lead to confusion and potential errors in your code. For this purpose, efforts have been made to effectively address this issue, both on names and action level.
+
+While zustand-slices currently doesn't throw errors for name collisions, using TypeScript is highly recommended. This allows for robust type-level detection of potential conflicts. The following sections present in detail the way collisions are handled in `withSlices` and `withActions` utilities.
+
+## withSlices
+
+### Slice names
+
+Collisions on slice names occur if they are shared within multiple slices. This includes avoiding conflicts with other slice and action names. As presented in the example below, `countSlice` and `anotherCountSlice` share the same name `count`, which causes the `combinedConfig` to be typed as `never`.
+
+```js
+const countSlice = createSlice({
+  name: 'count',
+  value: 0,
+  actions: {
+    inc: () => (prev) => prev + 1,
+  },
+});
+
+const anotherCountSlice = createSlice({
+  name: 'count',
+  value: 0,
+  actions: {
+    inc: () => (prev) => prev + 1,
+  },
+});
+
+const combinedConfig = withSlices(countSlice, anotherCountSlice);
+//        ^? const combinedConfig: never
+const useCountStore = create(combinedConfig);
+//        ^? const useCountStore: UseBoundStore<StoreApi<unknown>>
```

Comment:
Ah, I actually wasn't aware that it infers as `unknown`. Maybe I can revert to the previous type where it warns on the parameters.

Reply:
Yes, it would be nice if we could get a warning at that point of defining the hook. Is this related to Zustand or zustand-slices types?

Reply:
#16